packages/wasm-dpp/test/unit/dataContract/stateTransition/DataContractUpdateTransition/validation/state/validateDataContractUpdateTransitionStateFactory.spec.js

```diff
@@ -30,82 +39,87 @@ describe('validateDataContractUpdateTransitionStateFactory', () => {
 
     stateTransition = new DataContractUpdateTransition({
       dataContract: updatedRawDataContract,
+      protocolVersion: protocolVersion.latestVersion,
     });
 
     executionContext = new StateTransitionExecutionContext();
 
     stateTransition.setExecutionContext(executionContext);
 
-    stateRepositoryMock.fetchDataContract.resolves(dataContract);
+    const validator = new DataContractValidator();
+    const dataContractFactory = new DataContractFactory(protocolVersion.latestVersion, validator);
+    const wasmDataContract = await dataContractFactory.createFromBuffer(dataContract.toBuffer());
 
-    validateDataContractUpdateTransitionState = validateDataContractUpdateTransitionStateFactory(
-      stateRepositoryMock,
-    );
+    const stateRepositoryLike = {
+      fetchDataContract: () => wasmDataContract,
+    };
+
+    factory = (t) => validateDataContractUpdateTransitionState(stateRepositoryLike, t);
   });
 
   it('should return invalid result if Data Contract with specified contractId was not found', async () => {
-    stateRepositoryMock.fetchDataContract.resolves(undefined);
+    const stateRepositoryLikeNoDataContract = {
+      fetchDataContract: () => undefined,
+    };
+
+    const factoryNoDataContract = (t) => validateDataContractUpdateTransitionState(
+      stateRepositoryLikeNoDataContract,
+      t,
+    );
 
-    const result = await validateDataContractUpdateTransitionState(stateTransition);
+    const result = await factoryNoDataContract(stateTransition);
 
-    expectValidationError(result, DataContractNotPresentError);
+    expect(result.isValid()).to.be.false();
 
     const [error] = result.getErrors();
 
+    expect(error).to.be.an.instanceOf(DataContractNotPresentError);
     expect(error.getCode()).to.equal(1018);
-    expect(Buffer.isBuffer(error.getDataContractId())).to.be.true();
     expect(error.getDataContractId()).to.deep.equal(dataContract.getId().toBuffer());
-
-    expect(stateRepositoryMock.fetchDataContract).to.be.calledOnceWithExactly(
-      dataContract.getId(),
-      executionContext,
-    );
   });
 
   it('should return invalid result if Data Contract version is not larger by 1', async () => {
-    dataContract.version -= 1;
+    const badlyUpdatedRawDataContract = dataContract.toObject();
+    badlyUpdatedRawDataContract.version += 2;
 
-    const result = await validateDataContractUpdateTransitionState(stateTransition);
+    const badStateTransition = new DataContractUpdateTransition({
+      dataContract: badlyUpdatedRawDataContract,
+      protocolVersion: protocolVersion.latestVersion,
+    });
 
-    expectValidationError(result, InvalidDataContractVersionError);
+    const result = await factory(badStateTransition);
```

Comment:
Let's give it a more descriptive name as well, something like `validateDataContractUpdateTransitionState` maybe?